Android 16KB ページサイズ対応。
Google Play Store が Android 15 以降の端末向けに要求する 16KB アライメントに
対応するためのリンカフラグを追加する。
参照: https://developer.android.com/guide/practices/page-sizes

このパッチは disable-cparam-search.patch 適用後に当てること
(コンテキスト行に USE_CPARAM_SEARCH 0 が含まれるため)。

diff --git a/Source/CMakeLists.txt b/Source/CMakeLists.txt
--- a/Source/CMakeLists.txt
+++ b/Source/CMakeLists.txt
@@ -184,7 +184,9 @@ elseif(ARX_TARGET_PLATFORM_ANDROID)
 
     set(USE_CPARAM_SEARCH 0)
     set(ARX_INSTALL_LIBRARY_DIR "lib/${ANDROID_ABI}")
+    # [Covelline] Android 16KB ページサイズ対応フラグ
+    set(CMAKE_SHARED_LINKER_FLAGS "${CMAKE_SHARED_LINKER_FLAGS} -Wl,-z,max-page-size=16384")
     # Take only the first 3 fields from ANDROID_TOOLCHAIN_NAME (i.e. drop the "clang" or gcc version suffix, if present).
     string(REGEX MATCH "^[^-]+-[^-]+-[^-]+" ARX_ANDROID_SYSTEM_LIBDIR_NAME ${ANDROID_TOOLCHAIN_NAME})
     set(ARX_ANDROID_SYSTEM_LIBDIR ${CMAKE_SYSROOT}/usr/lib/${ARX_ANDROID_SYSTEM_LIBDIR_NAME})
